packages/babel-traverse/test/scope.js

```diff
@@ -250,4 +271,84 @@ describe("scope", function() {
       });
     });
   });
+
+  describe("duplicate bindings", () => {
+    /*
+     * These tests do not use the parser as the parser has
+     * its own scope tracking and we want to test the scope tracking
+     * of traverse here and see if it handles duplicate bindings correctly
+     */
+    describe("catch", () => {
+      // try {} catch (e) { let e; }
+      const createTryCatch = function(kind) {
+        return t.tryStatement(
+          t.blockStatement([]),
+          t.catchClause(
+            t.identifier("e"),
+            t.blockStatement([
+              t.variableDeclaration(kind, [
+                t.variableDeclarator(t.identifier("e"), t.stringLiteral("1")),
+              ]),
+            ]),
+          ),
+        );
+      };
+      ["let", "const"].forEach(name => {
+        it(name, () => {
+          const ast = createTryCatch(name);
+
+          expect(() => getPath(ast)).toThrowErrorMatchingSnapshot();
+        });
+      });
+
+      it("var", () => {
+        const ast = createTryCatch("var");
+
+        expect(getPath(ast).node).toMatchSnapshot();
```

Comment:
Nit: `generator(getPath(ast).node)` would make the snapshot output easier to check.

Reply:
I was thinking exactly the same, I guess what I'm really testing here is that it does not throw. So `expect().not.toThrow()` makes sense to me.